File: HackerRank/30 Days of Code/Day1_DataTypes.cs
// Day 1: Data Types
// https://www.hackerrank.com/challenges/30-data-types/problem

using System;
using System.Collections.Generic;
using System.IO;

class Solution {
    static void Main(String[] args) {
        int i = 4;
        double d = 4.0;
        string s = "HackerRank ";

        int j = int.Parse(Console.ReadLine());
        double e = double.Parse(Console.ReadLine());
        string t = Console.ReadLine();
        Console.WriteLine(i + j);
        Console.WriteLine("{0:f1}", d + e);
        Console.WriteLine(s + t);
    }
}


// 정수형, 실수형, 문자열을 입력받고 미리 만들어진 변수와 더하기/붙이기
// C#에서 실수형 출력 정확도 설정은 저러한 포매팅 방식으로 사용(f는 고정 소수점)

File: HackerRank/30 Days of Code/Day1_DataTypes.java
// Day 1: Data Types
// https://www.hackerrank.com/challenges/30-data-types/problem

import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {
	
    public static void main(String[] args) {
        int i = 4;
        double d = 4.0;
        String s = "HackerRank ";
		
        Scanner scan = new Scanner(System.in);

        int j = scan.nextInt();
        double e = scan.nextDouble();
        scan.nextLine();
        String t = scan.nextLine();
        
        System.out.println(i + j);
        System.out.println(d + e);
        System.out.println(s + t);

        scan.close();
    }
}

// 정수형, 실수형, 문자열을 입력받고 미리 만들어진 변수와 더하기/붙이기
// Scanner.nextLine() 사용 전에 nextLine()을 쓰지 않았다면 빈 nextLine()으로 플러싱
// 여기는 실수형 결과에 자동으로 .0을 붙여줌